src/main/java/com/easypost/model/Rate.java

```diff
@@ -3,7 +3,7 @@
 import com.easypost.exception.EasyPostException;
 import com.easypost.net.EasyPostResource;
 
-public final class Rate extends EasyPostResource {
```

Comment:
You can't inherit a class that is `final`

Reply:
Right - that's the why.

I'd like to know what changes because this class is no longer final?

Reply:
We removed the `final` from the Rate class so the `PickupRate` class can inherit it. The main reason for doing this is for the type casting between `Rate` and `PickupRate`, so we don't need to write a separate `lowestRate` function for `PickupRate`.

Reply:
Because the compiler doesn't have to worry about re-referencing a `final` class or variable, since something that is `final` cannot be overwritten/extended, there is a [slight performance improvement](https://www.baeldung.com/java-final-performance) when things are declared `final`. But in the grand scheme of things, the benefits (extending this class) outweigh the risks (realistically, there are be no negative repercussions to not declaring this class `final`).